src/sentry/snuba/metrics/datasource.py

```diff
@@ -176,65 +176,126 @@ def _validate_requested_derived_metrics_in_input_metrics(
         )
 
 
-def get_single_metric_info(projects: Sequence[Project], metric_name: str) -> MetricMetaWithTagKeys:
-    assert projects
-
+def _fetch_tags_or_values_per_ids(
+    projects: Sequence[Project],
+    metric_names: Optional[Sequence[str]],
+    referrer: str,
+    column: str = "tags.key",
+) -> Tuple[Sequence[Union[Tag, TagValue]], Optional[str]]:
```

Comment:
Is it not the same thing?

Reply:
"sequence of union" would allow you to return something like `[Tag(..), Tag(..), TagValue(..)]`. But it should be either all `Tag` or all `TagValue`, right?

Reply:
Ah right!